spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/query/JdbcQueryCreator.java

```diff
@@ -177,13 +178,23 @@ protected ParametrizedQuery complete(@Nullable Criteria criteria, Sort sort) {
 			completedBuildSelect = selectOrderBuilder.lock(this.lockMode.get().value());
 		}
 
-		Select select = completedBuildSelect.build();
+		Select select = getSelect(completedBuildSelect);
 
 		String sql = SqlRenderer.create(renderContextFactory.createRenderContext()).render(select);
 
 		return new ParametrizedQuery(sql, parameterSource);
 	}
 
+	private Select getSelect(SelectBuilder.BuildSelect completedBuildSelect) {
```

Comment:
Why is the selection list empty in the first place? I'd assume that we pass-thru `ReturnedType.InputProperties` without checking whether these are empty.